### PR TITLE
Return resource version in the header

### PR DESCRIPTION
**Pull request by e0ipso**
When one does `api/articles` they should know which resource's version served their response.

So we can return the version for all cases (e.g. also in `api/v1.0/articles`)

### CODE REVIEW

**Comment by e0ipso** [comment]
I also took the opportunity to change our custom version header to use something more widely spread:

`X-Restful-Version` by `X-API-Version`.

https://www.npmjs.org/package/restful-versioning

File: tests/RestfulViewEntityTestCase.test [hunk 1]
```diff
@@ -274,10 +274,16 @@ class RestfulViewEntityTestCase extends RestfulCurlBaseTestCase {
...
-    $response = $this->httpRequest('api/articles', \RestfulInterface::GET, NULL, array('X-Restful-Version' => 'v1.1'));
-    $this->assertTrue(preg_match('/X-Restful-Version/', $response['headers']), 'The Vary header was added.');
+    // Make sure that the version header is present in the response.
+    $this->assertTrue(preg_match('/v1\.1/', $response['headers']), 'The Vary header was added.');
+
+    $response = $this->httpRequest('api/articles', \RestfulInterface::GET, NULL, array('X-API-Version' => 'v1.1'));
+    $this->assertTrue(preg_match('/X-API-Version/', $response['headers']), 'The Vary header was added.');
+
+    // Make sure that the version header is present in the response.
+    $this->assertTrue(preg_match('/v1\.1/', $response['headers']), 'The Vary header was added.');
```
Comment (amitaibu) [inline tests/RestfulViewEntityTestCase.test:286]: `The Vary header was added.` => `The version header is present in the response.`

**Comment by amitaibu** [comment]
Looks good, with just minor comment. Once green, can be merged.

File: tests/RestfulViewEntityTestCase.test (hunk 1 above)
Comment (e0ipso) [inline tests/RestfulViewEntityTestCase.test:286]: Actually this is intentional.

The `$response['headers']` only contains the contents of the header, so in this case if we want to detect `Vary: X-API-Version` we need to find `'X-API-Version'`. On the other hand if we want to detect `X-API-Version: v1.1` we'll look for `'v1.1'`.

**Pull request merged**